tablo/csv_utils.py

```diff
@@ -140,25 +166,16 @@ def apply_headers(row_set, row):
     return apply_headers
 
 
-class EmptyType(CellType):
-    """ An empty cell """
-
-    result_type = unicode_string
-    guessing_weight = 0
-
-    def cast(self, value):
-        if value is None:
-            return None
-        if isinstance(value, self.result_type):
-            return value
-        try:
-            return unicode_string(value)
-        except UnicodeEncodeError:
-            return str(value)
+def name_for_type(class_or_instance):
 
+    if isinstance(class_or_instance, type):
+        type_name = TYPE_NAMES.get(class_or_instance)
+    elif isinstance(class_or_instance, CellType):
+        type_name = TYPE_NAMES.get(class_or_instance.__class__)
+    elif isinstance(class_or_instance, Cell):
+        type_name = TYPE_NAMES.get(class_or_instance.type.__class__)
```

Comment:
The `TYPE_NAMES` mapping it unnecessary. You can instead do:

```python
type_name = class_or_instance.__class__.__name__
```

... and so on.

Reply:
The `TYPE_NAMES` constant removes "Type" from each of them. So, for instance:
```python
TYPES = (EmptyType, StringType, DecimalType, IntegerType, DateType)
TYPE_NAMES = {
    IntegerType: 'Integer',
    DateType: 'Date',
    DecimalType: 'Decimal',
    StringType: 'String',
    EmptyType: 'Empty'
}
```
The `TYPE_NAMES` constant is there to:

1. Represent only supported type names (otherwise `?`)
2. Present them in a standard way (without `'*Type'`)

If I were just to do `class_or_instance.__class__.__name__` I get the "Type" value appended.

Also, if I do `str(class_or_instance)`, then even though I would get `'Integer'` for `IntegerType`, I will get `'Date(None)'` for `DateType`. This is just a simple helper to make `tablo` return standard values for `messytables.types` in raised exceptions.

Reply:
Ah, ok. I'd missed that.